providers/implementations/rands/seed_src_jitter.c

```diff
@@ -0,0 +1,322 @@
+    s->state = EVP_RAND_STATE_UNINITIALISED;
+    return s;
+}
+
+static void jitter_free(void *vseed)
+{
+    OPENSSL_free(vseed);
+}
+
+static int jitter_instantiate(void *vseed, unsigned int strength,
+                              int prediction_resistance,
+                              const unsigned char *pstr,
+                              size_t pstr_len,
+                              ossl_unused const OSSL_PARAM params[])
+{
+    PROV_JITTER *s = (PROV_JITTER *)vseed;
+
+    if (jent_entropy_init_ex(0, JENT_FORCE_FIPS))
+        return 0;
+
+    s->state = EVP_RAND_STATE_READY;
+    return 1;
+}
+
+static int jitter_uninstantiate(void *vseed)
+{
+    PROV_JITTER *s = (PROV_JITTER *)vseed;
+
+    s->state = EVP_RAND_STATE_UNINITIALISED;
+    return 1;
+}
+
+static int jitter_generate(void *vseed, unsigned char *out, size_t outlen,
+                           unsigned int strength,
+                           ossl_unused int prediction_resistance,
+                           ossl_unused const unsigned char *adin,
+                           ossl_unused size_t adin_len)
+{
+    PROV_JITTER *s = (PROV_JITTER *)vseed;
+    size_t entropy_available;
+    RAND_POOL *pool;
+
+    if (s->state != EVP_RAND_STATE_READY) {
+        ERR_raise(ERR_LIB_PROV,
+                  s->state == EVP_RAND_STATE_ERROR ? PROV_R_IN_ERROR_STATE
+                                                   : PROV_R_NOT_INSTANTIATED);
+        return 0;
+    }
+
+    pool = ossl_rand_pool_new(strength, 1, outlen, outlen);
+    if (pool == NULL) {
+        ERR_raise(ERR_LIB_PROV, ERR_R_RAND_LIB);
+        return 0;
+    }
+
+    /* Get entropy from jitter entropy library. */
+    entropy_available = ossl_prov_acquire_entropy_from_jitter(pool);
+
+    if (entropy_available > 0)
+        memcpy(out, ossl_rand_pool_buffer(pool), ossl_rand_pool_length(pool));
+
+    ossl_rand_pool_free(pool);
+    return entropy_available > 0;
+}
+
+static int jitter_reseed(void *vseed,
+                         ossl_unused int prediction_resistance,
+                         ossl_unused const unsigned char *ent,
+                         ossl_unused size_t ent_len,
+                         ossl_unused const unsigned char *adin,
+                         ossl_unused size_t adin_len)
+{
+    PROV_JITTER *s = (PROV_JITTER *)vseed;
+
+    if (s->state != EVP_RAND_STATE_READY) {
+        ERR_raise(ERR_LIB_PROV,
+                  s->state == EVP_RAND_STATE_ERROR ? PROV_R_IN_ERROR_STATE
+                                                   : PROV_R_NOT_INSTANTIATED);
+        return 0;
+    }
+    return 1;
+}
+
+static int jitter_get_ctx_params(void *vseed, OSSL_PARAM params[])
+{
+    PROV_JITTER *s = (PROV_JITTER *)vseed;
+    OSSL_PARAM *p;
+
+    p = OSSL_PARAM_locate(params, OSSL_RAND_PARAM_STATE);
+    if (p != NULL && !OSSL_PARAM_set_int(p, s->state))
+        return 0;
+
+    p = OSSL_PARAM_locate(params, OSSL_RAND_PARAM_STRENGTH);
+    if (p != NULL && !OSSL_PARAM_set_int(p, 1024))
```

Comment:
This seems high. But I see SEED-SRC has the same...

Reply:
@mattcaswell i'm not sure any of these parameters have any meaning. I chose to copy them as is, to maintain drop-in bug for bug compatibility with SEED-SRC. I hope we can consider separately if these should be removed/changed in jitter or seed-src or both separately from trying to add jitter. As I think this part of openssl is getting attention, for the first time in a little while.

Reply:
I'd be interested in @paulidale's view on this topic - but I don't consider this to be a blocker for this PR.

Reply:
Way back I picked a number that was sufficiently high so that it would never be a problem in the future (famous last words).  Since this is a seed source, it can produce entropy to meet any desired level of strength so the specific number isn't important, it just has to be greater than or equal to the maximum strength of everything downstream.  There are a few places where this is used (DRBG parenting e.g.) but it's not widespread.

I don't think this should be addressed here.  Better would be an OTC discussion followed by a new PR if required.

Reply:
I agree with @paulidale above, that is my reading as well. With all the specs currently there are justifications of minimum entropy per byte, and minimum requirements to request at least N, where N so far has not been more than 256. And yes the current number is sufficiently high, but also a reasonable ceiling.

Marking this as resolved. If we need to schedule a followup about this, @mattcaswell please open an issue to track that.

Reply:
I'm happy with the responses on this. No follow up needed.